Improve focus handling for form field visibility

fahrigedik opened the pull request:
Refactored focus logic to ensure the field is only focused when visible. Added IntersectionObserver to handle cases where the element is not immediately visible, improving accessibility and user experience.

<img width="963" height="826" alt="image" src="https://github.com/user-attachments/assets/abad1c81-b9c8-4104-a2fd-45948d05dcc4" />

### Description

Resolves https://github.com/volosoft/vs-internal/issues/7593 (write the related issue number if available)

### Checklist

- [X] I fully tested it as developer / designer and created unit / integration tests
- [X] I documented it (or no need to document or I will create a separate documentation issue)

### How to test it?

you have to change branch rel-9.3 on volo
you have to change branch issue-7593 on abp
you have to change branch rel-4.3 on lepton
and you can run commercial test app on vs-internal

fahrigedik committed:
Improve focus handling for form field visibility

Refactored focus logic to ensure the field is only focused when visible. Added IntersectionObserver to handle cases where the element is not immediately visible, improving accessibility and user experience.

sumeyyeKurtulus reviewed (commented):
Hello @fahrigedik thank you for checking this problem in the forms. However, we can call the animation frame request instead which is more straightforward and clean
```tsx
 requestAnimationFrame(() => {
    this.fieldRef.nativeElement.focus();
 });
```

fahrigedik committed:
Simplify initial field focus logic in form prop component

Replaces the IntersectionObserver-based focus logic with a simpler requestAnimationFrame approach for focusing the first field. This reduces complexity and potential timing issues when focusing the field after view initialization.
